src/bin/triplydb-create-query-job.ts

```diff
@@ -107,29 +89,42 @@ const command = program
     const account = await app.getUser(options.account);
     // check whether account name exists
     await account.getInfo();
-    const queries: { queryId: string; priority?: number }[] = [];
-    for (const query of queryInfo) {
-      const queryAccount = await app.getAccount(query.queryAccountName);
-      const queryId = (await (await queryAccount.getQuery(query.queryName)).getInfo()).id;
-      queries.push({ queryId, priority: query.priority });
-    }
-    const sourceDatasetAccount = await app.getAccount(sourceDatasetAccountName);
-    const sourceDatasetId = (await (await sourceDatasetAccount.getDataset(sourceDatasetName)).getInfo()).id;
-    const targetDatasetAccount = await app.getAccount(targetDatasetAccountName);
-    const targetDatasetId = (await (await targetDatasetAccount.ensureDataset(targetDatasetName)).getInfo()).id;
-    const payload: QueryJobPipelineCreate = {
-      queries: queries,
-      sourceDatasetId: sourceDatasetId,
-      targetDatasetId: targetDatasetId,
-      targetGraphName: options.targetGraphName,
-    };
-    const queryJob: QueryJob = new QueryJob(app, account);
-    try {
-      await queryJob.createQueryJobPipeline(payload, queryInfo);
-    } catch (e) {
-      console.error(e);
-      process.exit(1);
-    }
+    readFile(configFile[0], async function (err, data) {
+      if (err) sanityCheckError(err.message);
+      if (data) {
+        let queryJobConfig: QueryJobPipelineCreate | undefined = undefined;
+        try {
+          queryJobConfig = JSON.parse(data.toString()) as QueryJobPipelineCreate;
+        } catch (e) {
+          console.error(e);
+          process.exit(1);
+        }
+        if (queryJobConfig) {
+          if ("version" in queryJobConfig) {
+            const queryInfo: QueryInformation[] = [];
+            if ("queries" in queryJobConfig && queryJobConfig.queries) {
+              for (const query of queryJobConfig.queries as any[]) {
+                const [queryAccountName, actualQueryName] = query.name.split("/");
+                if (!queryAccountName) sanityCheckError(`Missing query account name for query "${query.name}"`);
+                if (!actualQueryName) sanityCheckError(`Missing query name for query "${query.name}"`);
+                queryInfo.push({
+                  queryAccountName: queryAccountName,
+                  queryName: actualQueryName,
+                  priority: query.priority || 0,
+                });
+              }
+            }
+            const queryJob: QueryJob = new QueryJob(app, account);
```

Comment:
I'm confused by the amount of code involved. If I understand correctly, the API supports getting the whole pipeline definition as payload in the body (and return an appropriate message when it's invalid). Why would we need to add custom validation and parsing?

Reply:
Hmm, ignore this comment, I think we should first discuss (in a separate ticket) how we want to include pipelines and jobs in triplydb-js (from the JS api perspective). We can then at a later stage clean this code up